cloud/services/publicloadbalancers/publicloadbalancers.go

```diff
@@ -72,89 +76,94 @@ func (s *Service) Reconcile(ctx context.Context, spec interface{}) error {
-							},
-							Probe: &network.SubResource{
-								ID: to.StringPtr(fmt.Sprintf("/%s/%s/probes/%s", idPrefix, lbName, probeName)),
-							},
+						},
+						BackendAddressPool: &network.SubResource{
+							ID: to.StringPtr(fmt.Sprintf("/%s/%s/backendAddressPools/%s", idPrefix, lbName, backEndAddressPoolName)),
 						},
 					},
 				},
-				OutboundRules: &[]network.OutboundRule{
-					{
-						Name: to.StringPtr("OutboundNATAllProtocols"),
-						OutboundRulePropertiesFormat: &network.OutboundRulePropertiesFormat{
-							Protocol:             network.LoadBalancerOutboundRuleProtocolAll,
-							IdleTimeoutInMinutes: to.Int32Ptr(4),
-							FrontendIPConfigurations: &[]network.SubResource{
-								{
-									ID: to.StringPtr(fmt.Sprintf("/%s/%s/frontendIPConfigurations/%s", idPrefix, lbName, frontEndIPConfigName)),
-								},
-							},
-							BackendAddressPool: &network.SubResource{
-								ID: to.StringPtr(fmt.Sprintf("/%s/%s/backendAddressPools/%s", idPrefix, lbName, backEndAddressPoolName)),
-							},
-						},
+			},
+		},
+	}
+
+	if publicLBSpec.Role == infrav1.APIServerRole {
+		probeName := "tcpHTTPSProbe"
+		lb.LoadBalancerPropertiesFormat.Probes = &[]network.Probe{
+			{
+				Name: to.StringPtr(probeName),
+				ProbePropertiesFormat: &network.ProbePropertiesFormat{
+					Protocol:          network.ProbeProtocolTCP,
+					Port:              to.Int32Ptr(s.Scope.APIServerPort()),
+					IntervalInSeconds: to.Int32Ptr(15),
+					NumberOfProbes:    to.Int32Ptr(4),
+				},
+			},
+		}
+		// We disable outbound SNAT explicitly in the HTTPS LB rule and enable TCP and UDP outbound NAT with an outbound rule.
+		// For more information on Standard LB outbound connections see https://docs.microsoft.com/en-us/azure/load-balancer/load-balancer-outbound-connections.
+		lb.LoadBalancerPropertiesFormat.LoadBalancingRules = &[]network.LoadBalancingRule{
+			{
+				Name: to.StringPtr("LBRuleHTTPS"),
+				LoadBalancingRulePropertiesFormat: &network.LoadBalancingRulePropertiesFormat{
+					DisableOutboundSnat:  to.BoolPtr(true),
+					Protocol:             network.TransportProtocolTCP,
+					FrontendPort:         to.Int32Ptr(s.Scope.APIServerPort()),
+					BackendPort:          to.Int32Ptr(s.Scope.APIServerPort()),
+					IdleTimeoutInMinutes: to.Int32Ptr(4),
+					EnableFloatingIP:     to.BoolPtr(false),
+					LoadDistribution:     network.LoadDistributionDefault,
+					FrontendIPConfiguration: &network.SubResource{
+						ID: to.StringPtr(fmt.Sprintf("/%s/%s/frontendIPConfigurations/%s", idPrefix, lbName, frontEndIPConfigName)),
+					},
+					BackendAddressPool: &network.SubResource{
+						ID: to.StringPtr(fmt.Sprintf("/%s/%s/backendAddressPools/%s", idPrefix, lbName, backEndAddressPoolName)),
+					},
+					Probe: &network.SubResource{
+						ID: to.StringPtr(fmt.Sprintf("/%s/%s/probes/%s", idPrefix, lbName, probeName)),
 					},
 				},
 			},
-		})
+		}
+	}
+
+	err = s.Client.CreateOrUpdate(ctx,
```

Comment:
nit: not sure what the formatting rules are but this feels like it should be on a single line

Reply:
fixed, this was leftover from when lb was the huge struct definition